tests/shared/src/test/scala-3/cats/tests/FunctionKLiftSuite.scala

```diff
@@ -34,4 +34,12 @@ class FunctionKLiftSuite extends CatsSuite {
       assert(fHeadOption(a) === a.headOption)
     }
   }
+
+  test("lift a function directly using Scala 2 compatible syntax") {
+    def headOption[A](list: List[A]): Option[A] = list.headOption
+    val fHeadOption = FunctionK.liftFunction[List, Option](headOption)
+    forAll { (a: List[Int]) =>
+      assert(fHeadOption(a) === a.headOption)
+    }
+  }
```

Comment:
Just to note: I feel this test case should be extracted into a new test suite that would cross-compile for both Scala2 and Scala3. Looks like there's no such test suite available for `FunctionK`. We can do it in a separate follow up PR though.

Reply:
Ooh that's a good idea